pipenv/shells.py

```diff
@@ -228,12 +228,19 @@ def _detect_emulator():
         keys.append("cmder")
     if os.environ.get("MSYSTEM"):
         keys.append("msys")
+    # VS Code internal terminal support
+    if os.environ.get("TERM_PROGRAM") == "vscode":
+        keys.append("vscode") 
     return ",".join(keys)
 
 
 def choose_shell(project):
-    emulator = project.PIPENV_EMULATOR.lower() or _detect_emulator()
-    type_, command = project.detect_info()
+    emulator = "none"
+    try:
+        emulator = _detect_emulator() or project.PIPENV_EMULATOR.lower()
```

Comment:
Why change the order of the conditions

Reply:
Not sure since `PIPENV_EMULATOR` appears to be the override for auto-detection, but this also has conflict with base now since master is `project.s.PIPENV_EMULATOR.lower()`

Reply:
`PIPENV_EMULATOR` is for override of course